### PR TITLE
Converter of TOA reflectance from VIIRS/JPSS1 Moderate Resolution 6-Min L1B Swath 750m

### PR DESCRIPTION
**Pull request by weiwilliam**
## Description
This PR adds the converter of NASA's VIIRS L1b data.
It combines the two inputs of observation and geolocation.
Observation data (VJ102MOD): https://ladsweb.modaps.eosdis.nasa.gov/missions-and-measurements/products/VJ102MOD
Geolocation data (VJ103MOD): https://ladsweb.modaps.eosdis.nasa.gov/missions-and-measurements/products/VJ103MOD
NASA's VIIRS L1b User Guide is [here](https://ladsweb.modaps.eosdis.nasa.gov/api/v2/content/archives/Document%20Archive/Science%20Data%20Product%20Documentation/NASA_VIIRS_L1B_UG_August_2021.pdf).

The converter has passed the coding norm test.
The test of ioda-validate.x returns no error with the testoutput.

## Dependencies
No dependencies

## Impact
No impacts

## Checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

**Comment by weiwilliam** [comment]
Geometry variables are not converted, I will add it in later commits.

**Comment by weiwilliam** [comment]
Hi @BenjaminRuston @jeromebarre @PatNichols, I think this PR is ready for review.

**Review by jeromebarre — APPROVED** [review]
Looks good, thanks for adding this.

File: src/compo/viirs_l1bnc2ioda.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+    # get command line arguments
+    # Usage: python blah.py -i /path/to/obs/2021060801.nc /path/to/obs/2021060802.nc ... -t Analysis_time /path/to/obs/2021060823.nc
+    # -o /path/to/ioda/20210608.nc
+    # where the input obs could be for any desired interval to concatenated together. Analysis time is generally the midpoint of
+    # analysis window.
+    parser = argparse.ArgumentParser(
+        description=('Read NASA VIIRS M-band Level 1b file(s) and Converter'
+                     ' of native NetCDF format for observations of TOA reflectance'
+                     ' from VIIRS to IODA NetCDF format.')
```
Comment (jeromebarre) [inline src/compo/viirs_l1bnc2ioda.py]: can you add in the description where the data should be downloaded from?

Comment (weiwilliam) [reply src/compo/viirs_l1bnc2ioda.py]: Yes, I added them in.

**Comment by BenjaminRuston** [comment]
@PatNichols please check conventions

**Comment by jeromebarre** [comment]
@weiwilliam the coding norms test is failing 
1/336 Test   #1: iodaconv_compo_coding_norms .....................................................***Failed    0.62 sec

**Comment by weiwilliam** [comment]
> @weiwilliam the coding norms test is failing 1/336 Test #1: iodaconv_compo_coding_norms .....................................................***Failed 0.62 sec

Let me check, I did use the pycodestyle to see whether it is good. 
There are whitespaces on those lines I just added.

**Comment by PatNichols** [comment]
@weiwilliam is there a ctest for you code? I am not seeing it...

**Comment by PatNichols** [comment]
> @PatNichols please check conventions

@BenjaminRuston @gthompsnJCSDA I am fairly sure that the conventions were not followed on this one.

**Comment by PatNichols** [comment]
@BenjaminRuston @gthompsnJCSDA This is resuls for the ioda_validate on the output file:

[[0m[[1m[[34m  Warning: Attribute 'Location' is present but is not listed as a required or optional attribute in the spec.
[[0m[[1m[[34m  Warning: Attribute 'Channel' is present but is not listed as a required or optional attribute in the spec.
[[0m[[1m[[34m  Warning: Attribute 'Location' does not have a YAML spec.
[[0m[[1m[[34m  Warning: Attribute 'Channel' does not have a YAML spec.

**Comment by weiwilliam** [comment]
> @weiwilliam is there a ctest for you code? I am not seeing it...

Oh yeah, I forgot this. I keep thinking of tests in ufo is not there yet and forgot in here. Let me work on it.

**Comment by gthompsnJCSDA** [comment]
@weiwilliam  Do you want my assistance with this python script?  I can see some things to change.

**Comment by weiwilliam** [comment]
> @weiwilliam Do you want my assistance with this python script? I can see some things to change.

@gthompsnJCSDA Thank you, can you point me to changes you would like to have? Then I can learn more from it.

**Comment by weiwilliam** [comment]
Thank you, @gthompsnJCSDA! I'm testing the multiple inputs and I'll add the ctest stuff when we have decision on the folder arrangement.

**Review by jeromebarre — CHANGES_REQUESTED** [review]
The coding norms are still failing:

```
Start   1: iodaconv_compo_coding_norms
--
1620 | 1/336 Test   #1: iodaconv_compo_coding_norms .....................................................***Failed    0.62 sec

```
Run the test like so: `ctest -VV -R iodaconv_compo_coding_norms `
and look at the lines where it complains. Thanks!

**Comment by weiwilliam** [comment]
> The coding norms are still failing:
> 
> ```
> Start   1: iodaconv_compo_coding_norms
> --
> 1620 | 1/336 Test   #1: iodaconv_compo_coding_norms .....................................................***Failed    0.62 sec
> ```
> 
> Run the test like so: `ctest -VV -R iodaconv_compo_coding_norms ` and look at the lines where it complains. Thanks!

@jeromebarre  Yeah, I fixed the coding norms but have not pushed back yet. I'm adding the code to skip the nighttime files.

**Comment by weiwilliam** [comment]
Checks are passed.
Here is the last warning from ioda-validate.x
`Warning: Attribute 'description' is present but is not listed as a required or optional attribute in the spec.`
@gthompsnJCSDA told me he would like to add the 'description' attribute to IODA convention.

**Comment by gthompsnJCSDA** [comment]
@jeromebarre  Perhaps we should move ahead with this converter in the `compo` directory for now and change up the directory structure in some way at a future time.  While it isn't ideal to have a vis satellite image converter specifically within `compo` folder, the alternate of creating a `VIIRS` folder then makes it odd to find the previous composition-type data for VIIRS in compo instead of VIIRS.  What do you think about "kicking the can down the road" and dealing with re-organization later.

**Review by gthompsnJCSDA — APPROVED** [review]
Looks like all tests are passing now and all looks good to me.  I suggest we wait to a future date to reorganize directory structure, because I would like this one in particular to live in a different folder, not `compo` since visible satellite data/imagery is not really compo-only.  Thanks Shih-Wei.

**Review by jeromebarre — APPROVED** [review]

**Pull request merged**